sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/NuGetMetadataReference.cs

```diff
@@ -84,22 +64,25 @@ private static IPackageRepository CreatePackageRepository()
 
             return aggregateRepository;
         }
-
-        private static MetadataReference[] CreateReferences(string packageId, string packageVersion)
+        
+        public static MetadataReference[] Create(string packageId, string packageVersion)
         {
-            var matchingDlls = Directory.GetFiles(GetNuGetPackageDirectory(), "*.dll", SearchOption.AllDirectories)
+            EnsurePackageIsInstalled(packageId, packageVersion);
+
+            var packageDirectory = GetNuGetPackageDirectory(packageId, packageVersion);
+            var matchingDlls = Directory.GetFiles(packageDirectory, "*.dll", SearchOption.AllDirectories)
                 .Select(path => new FileInfo(path))
                 .GroupBy(file => file.Directory.Name)
                 .Where(group => allowedNugetLibDirectories.Any(y => group.Key.StartsWith(y)))
                 .OrderByDescending(group => group.Key)
                 .First();
 
             return matchingDlls.Select(file => MetadataReference.CreateFromFile(file.FullName)).ToArray();
```

Comment:
The documentation suggests caching for performance. I've removed all of the caching for now as the previous caching scheme was now more complicated than necessary since we no longer have an _AssemblyInitialize_. We can add some in later if required.